feat(site): initial scaffolding [WIP]

nothingismagick opened the pull request:
Just a start, needs a bit of text and TLC

Commit:
feat(site): initial scaffolding

Beanow commented:
Please be sure to commit / generate a file `/docs/CNAME` containing `tauri-apps.org`.
As per [this doc](https://help.github.com/en/articles/configuring-a-publishing-source-for-github-pages#publishing-your-github-pages-site-from-a-docs-folder-on-your-master-branch) it should be in the folder not root of the master branch.

Beanow commented on docs-generator/src/404.html:
```diff
@@ -0,0 +1,23 @@
+    sessionStorage.redirect = location.href;
+  </script>
+
+  <meta http-equiv="refresh" content="0;URL='/tauri'"></meta>
```
Looks like this wants to go to `/tauri` but should probably go to the root `/` instead.

nothingismagick replied:
ahhh - missed that one.
nice catch

Commit:
fix(redirect): 404

Beanow commented:
I'll leave the CNAME as an issue, since the DNS isn't set up for it right now anway.

Commit:
feat(CNAME): reference to tauri-apps.org

Commit:
feat(CNAME): add CNAME to build script CMD

Commit:
feat(work): various things, including patterns page

Commit:
feat(sidebar): Now scroll works

- needed to close the sidebar, because there is no docheight
- added a # to urls

Commit:
feat(slides): add landing carousel

Commit:
feat(landing): add ministories and roadmap and mermaid

Commit:
feat(links): update links, enhance mermaid

Commit:
feat(patterns): stylish page, 5 patterns, dynamic mermaid
- note: trollbridge -> api-bridge
- update mermaid

Commit:
feat(patterns):
- add multiwin pattern
- add pattern icons
- rename patterns
- add config section with dynamic markdown
- use keep-alive
- minor revisions to landing page

Commit:
feat(security): add docs/security page

Commit:
feat(patterns): add cloudish flowchart

Commit:
feat(patterns): fix infobar for mobile

Commit:
feat(quasar): update deps

Commit:
feat(examples): basic scaffolding

Commit:
feat(patterns): mosts, bests

Commit:
feat(patterns): icons

Commit:
feat(build): last manual build
